testing/mod.ts

```diff
@@ -140,49 +141,79 @@ async function createTestCase(
     }
   }
   if (previousPrinted(name, results)) {
-    report(result);
+    report(result, consoleLog);
   }
 }
 
 function initTestCases(
   stats: TestStats,
   results: TestResults,
   tests: TestDefinition[],
-  exitOnFail: boolean
+  exitOnFail: boolean,
+  consoleLog: logginFunctionType
 ): Array<Promise<void>> {
-  return tests.map(createTestCase.bind(null, stats, results, exitOnFail));
+  return tests.map(
+    createTestCase.bind(null, stats, results, exitOnFail, consoleLog)
+  );
 }
 
 async function runTestsParallel(
   stats: TestStats,
   results: TestResults,
   tests: TestDefinition[],
+  disableLog: boolean,
   exitOnFail: boolean
 ): Promise<void> {
+  let l, e;
   try {
-    await Promise.all(initTestCases(stats, results, tests, exitOnFail));
+    if (disableLog) {
+      l = console.log;
+      console.log = function() {};
+      console.error = function() {};
+    }
+    await Promise.all(initTestCases(stats, results, tests, exitOnFail, l));
   } catch (_) {
     // The error was thrown to stop awaiting all promises if exitOnFail === true
     // stats.failed has been incremented and the error stored in results
   }
+  if (disableLog) {
+    console.log = l;
+    console.error = e;
+  }
 }
 
 async function runTestsSerial(
   stats: TestStats,
   tests: TestDefinition[],
-  exitOnFail: boolean
+  exitOnFail: boolean,
+  disableLog: boolean
 ): Promise<void> {
   for (const { fn, name } of tests) {
     // See https://github.com/denoland/deno/pull/1452
     // about this usage of groupCollapsed
-    console.groupCollapsed(`test ${name} `);
+    let l, e;
+    if (disableLog) {
+      l = console.log;
+      e = console.error;
+      console.log = function() {};
+      console.error = function() {};
+    }
+    console.groupCollapsed("");
```

Comment:
Is missing name of group intentional?

Reply:
This was done before i haven't change it.